src/services/authService.js

```diff
@@ -38,52 +35,43 @@ class AuthService {
     this.firebase.initializeApp(this.config);
   }
 
-  loadUserProfile() {
-    return new Promise((resolve) => {
-      this.keycloak
-        .loadUserProfile()
-        .then((profile) => {
-          resolve(profile);
-        })
-        .catch(function () {
-          alert("Failed to load user profile");
-        });
-    });
-  }
-
   login = () => {
-    cfaSignIn("google.com").subscribe();
+    cfaSignIn("google.com").subscribe((user)=>{
+     this.setSession(user);
+    });
   };
 
   logout = () => {
-    cfaSignOut().subscribe();
-    this.setSession(null);
+    cfaSignOut().subscribe(()=>{
+     this.removeSession();
+    });
+    
   };
-
-  setSession = (accessToken) => {
-    if (accessToken) {
-      localStorage.setItem("accessToken", accessToken);
-      axios.defaults.headers.common.Authorization = `Bearer ${accessToken}`;
-    } else {
+  
+  
+  removeSession(){
       localStorage.removeItem("accessToken");
+      localStorage.removeItem("userDataKey");
       delete axios.defaults.headers.common.Authorization;
-    }
-  };
-
-  getAccessToken = () => localStorage.getItem("accessToken");
+      window.location.reload();
+  }
 
-  isValidToken = (accessToken) => {
-    if (!accessToken) {
-      return false;
-    }
+  setSession(user){
+    if (user) {
+      localStorage.setItem("accessToken", user.accessToken);
```

Comment:
Firebase IdToken are found through `await user.getIdToken()`.
This method fetches the updated idToken which is necessary to make request to server.

Reply:
> Firebase IdToken are found through `await user.getIdToken()`.
> This method fetches the updated idToken which is necessary to make request to server.

Okay Sure @Abhishek-kumar09  here I'm not setting up an access token after every state change so I need to go with the previous implementation with `await user.getIdToken()`.